arbiter-core/src/environment/instruction.rs

```diff
@@ -149,6 +149,9 @@ pub(crate) enum EnvironmentData {
 
     /// The query is for the balance of an account given by the inner `Address`.
     Balance(ethers::types::Address),
+
+    /// The query is for the nonce of an account given by the inner `Address`.
+    TransactionCount(ethers::types::Address),
```

Comment:
@0xJepsen 

See here. You added a new instruction when there is already a `Instruction::Query`. Yet, you used the `Outcome::QueryReturn` which is just a bit confusing.

If you want to make a new instruction, I think it's best to make a new outcome. If you want to have a query outcome, then I think you should use a query instruction.

Reply:
I don't understand what you mean. The transaction count defined here is as described in it's comments right here
```
/// [`EnvironmentData`] is an enum used inside of the [`Instruction::Query`] to
/// specify what data should be returned to the user.
/// Currently this may be the block number, block timestamp, gas price, or
/// balance of an account.
```
Which is exactly how it's used and why i used it the way i did. It's exactly the same as the implementation for the `get_gas_price and ()` and `get_block_number()`?